docs_src/security/Ch-Configuring-Add-On-Services.md

```diff
@@ -1 +1,189 @@
+the service key or hostname to the environment variable `ADD_SECRETSTORE_TOKENS`:
+
+```yaml
+...
+  secretstore-setup:
+    container_name: edgex-secretstore-setup
+    depends_on:
+    - security-bootstrapper
+    - vault
+    environment:
+      ADD_SECRETSTORE_TOKENS: 'device-camera'
+...
+
+```
+
+With that, `secretstore-setup` then can generate secretstore token from `Vault` and store it in
+the `TokenFile` path specified in the TOML configuration file like the above example.
+
+## Configure the secure Redis message bus to use
+
+This can be done in the EdgeX service `secretstore-setup` environment section of
+`Docker-compose` file to add the `redisdb` and the service key using it.
+
+So an example will be like `redisdb[device-virtual]` in which `redisdb` is the database resource name
+and the `device-virtual` is the service key or hostname and add that value into the environment variable
+`ADD_KNOWN_SECRETS`:
+
+```yaml
+...
+  secretstore-setup:
+    container_name: edgex-secretstore-setup
+    depends_on:
+    - security-bootstrapper
+    - vault
+    environment:
+      ADD_SECRETSTORE_TOKENS: 'device-camera, my-service'
+      ADD_KNOWN_SECRETS: redisdb[app-rules-engine],redisdb[device-rest],redisdb[device-virtual]
+...
+
+```
+
+
+## (Optional) Configure the ACL role of configuration/registry to use if the service depends on it
+
+This is a new step coming from `securing Consul` security features as part of EdgeX Ireland release.
+
+If the add-on service uses `Consul` as the configuration and/or registry service, then we also need to
+configure the environment variable `ADD_REGISTRY_ACL_ROLES` to tell `security-bootstrapper` to generate
+an ACL role for `Consul` to associate with its token.
+
+An example of configuring acl roles of the registry `Consul` for the add-on services
+`device-modbus` and `app-http-export` is like the following:
+
+```yaml
+...
+  consul:
+    container_name: edgex-core-consul
+    depends_on:
+    - security-bootstrapper
+    - vault
+    entrypoint:
+    - /edgex-init/consul_wait_install.sh
+    environment:
+      ADD_REGISTRY_ACL_ROLES: app-http-export,device-modbus
+...
+
+```
+
+## Configure the API gateway access route for add-on service
+
+If it is desirable to let user or other application services outside EdgeX's Docker network access
+the endpoint of the add-on service, then we can configure and add it via the environment variable of
+service `proxy-setup`, `ADD_PROXY_ROUTE`.  `proxy-setup` adds those services listed in that environment
+variable into the API gateway (aka Kong) route so that the endpoint can be accessible using Kong's proxy
+endpoint.
+
+One example of adding API gateway access route for service `device-camera` is give as follows:
+
+```yaml
+...
+edgex-proxy:
+      ...
+    environment:
+      ...
+      ADD_PROXY_ROUTE: "device-camera.http://device-camera:59985"
+      ...
+...
+```
+
+where port number `59985` is the internal port used by service `device-camera`.
```

Comment:
There is a little bit more going on here that needs to be explained, like the first part is a URL path, and the second part is a hostname, and maybe that we recommend they normally agree.

Reply:
ok, that details you would like to see is actually in [API Gateway documentation page](Ch-APIGateway.md) page, so I kind of briefly just mentioning here to avoid the duplication.

Reply:
added some brief explanation about the format of value